### PR TITLE
[utils] Add a util to help printing with colors. NFC

### PR DESCRIPTION
**Pull request by nkcsgexi**
Incorporating @DougGregor  and @akyrtzi 's suggestion, I have moved the utility to a more sharable place. 

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please test and merge

File: include/swift/Basic/ColorUtils.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines an 'OSColor' class for helping printing colorful outputs
+// to the terminal.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_COLOR_UTILS_H
+#define SWIFT_COLOR_UTILS_H
```
Comment (akyrtzi) [inline include/swift/Basic/ColorUtils.h]: The macro guards style for all `swift/Basic` headers is to prefix with `SWIFT_BASIC_`, and follow with no underscores, so change to `SWIFT_BASIC_COLORUTILS_H`.

File: include/swift/Basic/ColorUtils.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+#include "llvm/Support/raw_ostream.h"
+
+namespace swift {
+
+class OSColor {
```
Comment (akyrtzi) [inline include/swift/Basic/ColorUtils.h]: Add a doc-comment, e.g "RAII class for setting a color for a raw_ostream and resetting when it goes out-of-scope."

**Comment by nkcsgexi** [comment]
@swift-ci smoke test

**Comment by nkcsgexi** [comment]
@swift-ci smoke test

**Pull request merged**